src/dns/server.rs

```diff
@@ -74,20 +72,18 @@ impl Server {
     /// # Arguments
     ///
     /// * `addr` - The socket address on which to run the DNS server.
-    /// * `network` - The network of the current node.
     /// * `state` - The state of ztunnel.
     /// * `forwarder` - The forwarder to use for requests not handled by this server.
     #[allow(clippy::too_many_arguments)] // no good way of grouping arguments here..
-    pub async fn new<S: AsRef<str>>(
+    pub async fn new(
         domain: String,
         address: config::Address,
-        network: S,
         state: DemandProxyState,
         forwarder: Arc<dyn Forwarder>,
         metrics: Arc<Metrics>,
         drain: DrainWatcher,
         socket_factory: &(dyn SocketFactory + Send + Sync),
-        allow_unknown_source: bool,
```

Comment:
oh ok, this _was_ just for testing. I was wondering why it was `false` everywhere.

Reply:
Yeah I think we were just lazy in the tests basically/it was harder to get the source before